### ISSUE TITLE
Polymorphic schema clash with mongodb

### ISSUE THREAD

**Issue by MrBr** [paularmstrong/normalizr#445]
# Schema key is a reserved pathname in mongodb

I would like to have the same normalized model structure on the backend and the frontend. The issue I am getting is that I can't have schemes in the mongo that have a `schema` attribute and that causes an issue if I want to store polymorphic array in a mongo collection. Here is an example:

Normalizr polymorphic schema:
```
export const sectionSchema = new schema.Entity('sections');
export const sectionChildrenSchema = new schema.Array(
  {
    sections: sectionSchema,
    steps: model.stepSchema,
  },
  (value: SectionChild) => value.type,
);
```

Mongoose polymorphic schema:
```
const sectionOrStepSchema = new Schema({
  _id: Schema.Types.ObjectId,
  type: String // <---- THE ISSUE - can't have `schema` key here
});

const sectionsSchema = new Schema({
  children: [
    {
      type: sectionOrStepSchema,
      ref: (item: any) => item.type
    }
  ],
  type: { type: String, default: "sections" }
});
```

Basically I want to define schemes with the same shape both on the backend and frontend and I can't do it because `schema` key is reserved pathname in mongo and on the other side hardcoded in normalizr to be something different.

# Solution

The simplest solution would be to add a configurable `schema` key for normalized polymorphic items. I've already made a fork and test it. It works with a few changed lines.
